services/brig/test/integration/API/User/Account.hs

```diff
@@ -152,15 +157,19 @@ testCreateUserWithPreverified brig aws = do
                 "email" .= fromEmail e,
                 "email_code" .= c
               ]
-      usr <- postUserRegister reg brig
-      let uid = userId usr
-      get (brig . path "/self" . zUser uid) !!! do
-        const 200 === statusCode
-        const (Just e) === (userEmail <=< responseJsonMaybe)
-      liftIO $ Util.assertUserJournalQueue "user activate" aws (userActivateJournaled usr)
+      if Opt.setRestrictUserCreation (Opt.optSettings opts) == Just True
+        then do
+          postUserRegister' reg brig !!! const 403 === statusCode
+        else do
+          usr <- postUserRegister reg brig
+          let uid = userId usr
+          get (brig . path "/self" . zUser uid) !!! do
+            const 200 === statusCode
+            const (Just e) === (userEmail <=< responseJsonMaybe)
+          liftIO $ Util.assertUserJournalQueue "user activate" aws (userActivateJournaled usr)
   liftIO $ Util.assertEmptyUserJournalQueue aws
 
-testCreateUser :: Brig -> Galley -> Http ()
+testCreateUser :: Brig -> Galley -> Http () -- TODO: this has nothing to do with /register.  what's going on here?
```

Comment:
I think it got refactored into `randomUser` at some point. It may even have been made redundant with all the other tests, maybe we should verify this and delete it. Or, rename the test to ensure it is reflects the assertions which are about the conversations.

Reply:
I checked again, and `randomUser` is posting to `/i/user`, not to `/register`.

I don't really want to complicate this PR with a completely unrelated problem, and again I don't think this is a low-hanging fruit for us to attack now.  I just wanted to leave a note for the next person to pass by here that yes, this hasn't made a lot of sense in a while.